useIsShake.js

```diff
@@ -1,59 +1,40 @@
 import React, { useState, useEffect } from 'react';
 import { Accelerometer } from 'expo-sensors';
-import { QUOTES_IN_DURATION, QUOTES_IN_DELAY, QUOTES_OUT_DURATION, SVG_LOAD_DURATION, SVG_OUT_DURATION, SVG_IN_DURATION, SVG_IN_DELAY, TRANSITION_TIME_2 } from './constants';
 
 export function useIsShake() {
   const [{ x }, setData] = useState({
     x: 0,
   });
   const [subscription, setSubscription] = useState(null);
-  const [toggle, setToggle] = useState(false);
+  const [isShakeTriggered, setIsShakeTriggered] = useState(false);
+  const [isShakeReady, setIsShakeReady] = useState(false);
 
   let polls = [null, null];
   let diffs = [];
-  let shakeCount = 0;
-  let isToggleReady;
-  let hasToggledBefore;
 
   function isShake({ x: newX }) {
-    if (isToggleReady === true) {
-      if (diffs.length === 2) {
-        shakeCount++;
-        setToggle(true);
-        isToggleReady = false;
+    if (diffs.length === 2) {
+      setIsShakeTriggered(true);
+      setIsShakeReady(false);
 
-        if (hasToggledBefore === undefined) hasToggledBefore = false;
-        else if (hasToggledBefore === false) hasToggledBefore = true;
+      polls = [null, null];
+      diffs = [];
 
-        let debounceTime;
-        if (hasToggledBefore === false) {
-          debounceTime = Math.max(SVG_OUT_DURATION, (QUOTES_IN_DURATION + QUOTES_IN_DELAY));
-        } else {
-          debounceTime = TRANSITION_TIME_2;
-        }
-
-        const toReady = setTimeout(() => {
-          setToggle(false);
-          isToggleReady = true;
-          clearTimeout(toReady);
-        }, debounceTime);
```

Comment:
Debounce now being handled through useEffect call in hook and setTimeout call removed in favor of animation completion callbacks affecting update of state.